src/plugins/home/server/services/sample_data/routes/uninstall.ts

```diff
@@ -66,6 +66,40 @@ export function createUninstallRoute(
         ? context.dataSource.opensearch.legacy.getClient(dataSourceId).callAPI
         : context.core.opensearch.legacy.client.callAsCurrentUser;
 
+      let savedObjectsList = getFinalSavedObjects({
+        dataset: sampleDataset,
+        workspaceId,
+        dataSourceId,
+      });
+
+      /**
+       * Try to delete index patterns first to avoid partial delete issues.
+       * This is necessary because default index patterns cannot be deleted
+       * when a workspace is enabled and the user lacks the required permissions.
```

Comment:
> This is necessary because default index patterns cannot be deleted
> when a workspace is enabled and the user lacks the required permissions.

I might miss something here, do we have special check on index pattern when deleting?

Reply:
In this PR, we will delete index patterns of sample data first when user want to uninstall sample data. So there are different cases when user try to uninstall sample data.
- Case 1: Index pattern is default index pattern and user is `dashboards admin` or `workspace admin`, Index pattern can be removed
- Case 2: Index pattern is default index pattern and user is `workspace read and write` and `workspace read only`, index pattern can't be removed
- Case 3: Index pattern is not default index pattern and user is `dashboards admin`, `workspace admin` or `workspace read & write`, index pattern can be removed
- Case 4: Index pattern is not default index pattern and user is `workspace read only`, index pattern can't be removed
This PR will enhance case 1 and case 2, will try to remove ui settings first when deleted index patterns are default index pattern.